src/main/java/io/lettuce/core/AbstractRedisClient.java

```diff
@@ -297,6 +301,17 @@ protected void channelType(ConnectionBuilder connectionBuilder, ConnectionPoint
         }
     }
 
+    protected void resolver(ConnectionBuilder connectionBuilder, ConnectionPoint connectionPoint) {
+
+        LettuceAssert.notNull(connectionPoint, "ConnectionPoint must not be null");
+
+        if (connectionPoint.getSocket() == null) {
+            connectionBuilder.bootstrap().resolver(
+                    new DnsAddressResolverGroup(new DnsNameResolverBuilder().channelType(Transports.datagramChannelClass())
+                            .socketChannelType(Transports.socketChannelClass().asSubclass(SocketChannel.class))));
```

Comment:
[Supplemental Comments]
The key point of this PR. cf: https://netty.io/news/2016/05/26/4-1-0-Final.html
If we want to use non-blocking dns resolver, we need to pass `DnsAddressResolverGroup` to `Bootstrap`.

Also note that netty encourages users to support TCP fallback in Netty 4.1.37 release note:
https://netty.io/news/2019/06/28/4-1-37-Final.html

I followed Netty 4.1.37 release note, i.e., use `DnsNameResolverBuilder`  with `DatagramChannel` (UDP) and `SocketChannel` (TCP).

Since `Transports.socketChannelClass()` returns `Class<? extends Channel>` and `DnsNameResolverBuilder#socketChannelType` wants `Class<? extends SocketChannel> channelType`, 
we need to cast the return value using `asSubclass` (or change the return type of `Transports#socketChannelClass`).
I didn't attach a test code about this but I checked we can cast known socket channels safely like:
```java
    @Test
    void socketChannelClassCastTest() {
        NioSocketChannel.class.asSubclass(SocketChannel.class);
        KQueueSocketChannel.class.asSubclass(SocketChannel.class);
        EpollSocketChannel.class.asSubclass(SocketChannel.class);
    }
```